todo_app/app.py

```diff
@@ -1,34 +1,75 @@
 from flask.wrappers import Request
-from flask import Flask, render_template, redirect, url_for, request
-import backend
+from flask import Flask, render_template, redirect, url_for, request, jsonify
 
+import backend
+from flask_dance.contrib.github import github
+from flask_login import logout_user, login_required
 
 app = Flask(__name__)
 
+from models import db, login_manager
+from oauth import github_blueprint
+
+app.secret_key = 'notsosecretkey'
+app.register_blueprint(github_blueprint, url_prefix="/login")
+
+db.init_app(app)
+login_manager.init_app(app)
+
+with app.app_context():
+    db.create_all()
+
 @app.route("/",methods = ['POST', 'GET'])
 def index():
-    if request.method == 'POST':
+    if not github.authorized:
+        return redirect(url_for("github.login"))
+    
```

Comment:
We have this line repeatedly at the start of each function; it would be nice to be able to lose that & replace it with the intent rather than the implementation. One way would be to take advantage of the `login_required` decorator that you've imported, and add a handler for unauthorized users, something like:
```python
@login_manager.unauthorized_handler
def unauthed():
    return redirect(url_for("github.login"))

@app.route(...)
@login_required
def an_endpoint():
    # code
```